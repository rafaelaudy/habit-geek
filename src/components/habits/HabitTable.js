import React from "react";

import HabitRows from "./HabitRows";

const HabitTable = ({
  week,
  habits,
  isReadOnly,
  onUpdateHabit,
  toggleDayHabit
}) =>
  habits.length > 0 ? (
    <HabitRows
      habits={habits}
      week={week}
      isReadOnly={isReadOnly}
      toggleDayHabit={toggleDayHabit}
      onUpdateHabit={onUpdateHabit}
    ></HabitRows>
  ) : null;

export default HabitTable;
