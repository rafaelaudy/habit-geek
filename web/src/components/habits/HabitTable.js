import React from "react";

import HabitHeader from "./HabitHeader";
import HabitRows from "./HabitRows";

const HabitTable = ({
  week,
  habits,
  isReadOnly,
  onUpdateHabit,
  toggleDayHabit
}) =>
  habits.length > 0 ? (
    <div className="mb-3">
      <HabitHeader isReadOnly={isReadOnly} week={week}></HabitHeader>
      <HabitRows
        habits={habits}
        week={week}
        isReadOnly={isReadOnly}
        toggleDayHabit={toggleDayHabit}
        onUpdateHabit={onUpdateHabit}
      ></HabitRows>
    </div>
  ) : null;

export default HabitTable;
