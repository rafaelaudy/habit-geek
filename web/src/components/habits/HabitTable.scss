@import "../Variables.scss";

.habit {
  &__table {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
  }

  &__header-row,
  &__row {
    display: flex;

    &:last-child .habit__cell--today {
      border-bottom-width: $border-width;
      border-bottom-left-radius: 0.25rem;
      border-bottom-right-radius: 0.25rem;
    }
  }

  &__cell-habit-container {
    flex-grow: 1;
    display: flex;
    align-items: center;
  }

  &__cell-frequency-container {
    width: 380px;
    display: flex;
    justify-content: space-around;
  }

  &__cell-action-container {
    width: 44px;
    padding-bottom: 10px;
  }

  &__cell {
    width: 55px;
    text-align: center;

    &--today {
      border: $border-width solid $muted;
      border-width: 0 $border-width;
    }

    &--today-header {
      border: $border-width solid $muted;
      border-bottom-width: 0;
      margin-bottom: 0px;
      margin-top: -7px;
      padding-top: 5px;
    }

    .toggle {
      margin: 5px 0 5px 6px;
    }
  }
}

@media (max-width: 576px) {
  .habit {
    &__table {
      max-width: 400px;
      margin: auto;
    }

    &__row {
      flex-wrap: wrap;

      margin: auto;
      max-width: 380px;

      &:first-child .habit__cell-habit-container {
        height: 0;
      }
    }

    &__cell-habit-container {
      justify-content: center;
      height: 42px;
    }

    &__cell-frequency-container {
      order: 1;
    }

    &__cell {
      font-size: 1rem;
      padding-top: 5px;
      width: 46px;

      &--today {
        border: $border-width solid $muted;
        border-radius: 0.25rem;
        padding-top: 4px;
      }

      &--today-header {
        border: 0;
        margin: 0;
        padding-top: 5px;
      }
    }
  }
}
