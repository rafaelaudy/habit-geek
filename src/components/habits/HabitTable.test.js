import React from "react";
import { shallow } from "enzyme";
import HabitTable from "./HabitTable";

describe("HabitTable component", () => {
  it("Renders static elements", () => {
    const habitTable = shallow(
      <HabitTable
        week="y1w1"
        habits={[{}, {}]}
        isReadOnly={true}
        toggleDayHabit={() => {}}
        onUpdateHabit={() => {}}
      />
    );
    expect(habitTable).toMatchSnapshot();
  });

  it("Returns null if no habits is passed", () => {
    const habitTable = shallow(
      <HabitTable
        habits={[]}
        isReadOnly={true}
        toggleDayHabit={() => {}}
        onUpdateHabit={() => {}}
      />
    );
    expect(habitTable.type()).toBe(null);
  });
});
