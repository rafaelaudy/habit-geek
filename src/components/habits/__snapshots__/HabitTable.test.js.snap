// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitTable component Renders static elements 1`] = `
<HabitRows
  habits={
    Array [
      Object {},
      Object {},
    ]
  }
  isReadOnly={true}
  onUpdateHabit={[Function]}
  toggleDayHabit={[Function]}
  week="y1w1"
/>
`;
